### PR TITLE
Refactor WinHttpHandler test to use LoopbackServer.CreateClientAndServer

### PR DESCRIPTION
**Pull request by liveans**
Related to #96481

### CODE REVIEW

File: src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs [hunk 1]
```diff
@@ -100,36 +100,26 @@ public async Task SendAsync_SlowServerAndCancel_ThrowsTaskCanceledException()
...
+            TaskCompletionSource<bool> tcs = new TaskCompletionSource<bool>();
+            await LoopbackServer.CreateClientAndServerAsync(async uri =>
             {
-                var triggerResponseWrite = new TaskCompletionSource<bool>();
-                var triggerRequestWait = new TaskCompletionSource<bool>();
-
-                await LoopbackServer.CreateServerAsync(async (server, url) =>
+                using var client = new HttpClient(new WinHttpHandler());
+                HttpRequestException ex = await Assert.ThrowsAsync<HttpRequestException>(() => client.GetAsync(uri));
+                _output.WriteLine($"Exception: {ex}");
```
Comment (antonfirsov) [inline src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs:108]: Is this PR to observe the exception or could this refactor somehow mitigate the issue? If it's the former, unfortunately I don't think there is more we can see than what is already reported in the issue. :(

Note that #102452 *might* be related. Do we know if this happens against Framework or .NET 9 runs?

Comment (liveans) [reply src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs:108]: This PR just simplifies the code and removes unnecessary stuff. Not intended to fix it. I'll investigate this more.

**Review by MihaZupan — APPROVED** [review]

**Comment by liveans** [comment]
/ba-g Test failure is unrelated to change.

**Pull request merged**